tests/unit/mesh/test_pmesh.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+#include "unit_tests.hpp"
+
+namespace mfem
+{
+#ifdef MFEM_USE_MPI
+
+TEST_CASE("ParMeshGlobalIndices",  "[Parallel], [ParMesh]")
+{
+   const int ne = 5;
+
+   for (int dimension = 2; dimension < 4; ++dimension)
+   {
+      Mesh* mesh;
+      if (dimension == 2)
+      {
+         mesh = new Mesh(ne, ne, Element::QUADRILATERAL, 1, 1.0, 1.0);
+      }
+      else
+      {
+         mesh = new Mesh(ne, ne, ne, Element::HEXAHEDRON, 1, 1.0, 1.0,
+                         1.0);
+      }
+
+      ParMesh pmesh(MPI_COMM_WORLD, *mesh);
+
+      int globalN = 0;
+
+      enum EntityType { VERTEX, EDGE, FACE, ELEMENT };
+      // Loop over all types of mesh entities
+      for (int e=EntityType::VERTEX; e<=EntityType::ELEMENT; ++e)
+      {
+         if (e == EntityType::FACE && dimension == 2)
+         {
+            continue;
+         }
+
+         Array<HYPRE_Int> gi;
+
+         switch (e)
+         {
+            case EntityType::VERTEX:
+               globalN = mesh->GetNV();
+               pmesh.GetGlobalVertexIndices(gi);
+               break;
+            case EntityType::EDGE:
+               globalN = mesh->GetNEdges();
+               pmesh.GetGlobalEdgeIndices(gi);
+               break;
+            case EntityType::FACE:
+               globalN = mesh->GetNFaces();
+               pmesh.GetGlobalFaceIndices(gi);
+               break;
+            case EntityType::ELEMENT:
+               globalN = mesh->GetNE();
+               pmesh.GetGlobalElementIndices(gi);
+               break;
+         }
+
+         // Verify that global indices are unique on each process.
```

Comment:
This comment is confusing. It could be `Verify that the local entities don't share a global index`.